pkg/webhook/validation/validate.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/pingcap/chaos-mesh/api/v1alpha1"
+)
+
+const (
+	invalidConfigurationMsg = "invalid configuration"
+)
+
+var log = ctrl.Log.WithName("validate-webhook")
+
+// ValidateChaos handle the validation for chaos api
+func ValidateChaos(res *admissionv1beta1.AdmissionRequest, kind string) *admissionv1beta1.AdmissionResponse {
+	var err error
+	var permitted bool
+	var msg string
+	switch kind {
+	case "PodChaos":
+		permitted, msg, err = validatePodchaos(res.Object.Raw)
+	case "NetworkChaos":
+		permitted, msg, err = validateNetworkChaos(res.Object.Raw)
+	case "IoChaos":
+		permitted, msg, err = validateIoChaos(res.Object.Raw)
+	case "TimeChaos":
+		permitted, msg, err = validateTimeChaos(res.Object.Raw)
+	default:
+		log.Error(err, "Could not unmarshal raw object")
+	}
+	if err != nil {
+		return &admissionv1beta1.AdmissionResponse{
+			Result: &metav1.Status{
+				Message: err.Error(),
+			},
+		}
+	}
+
+	return &admissionv1beta1.AdmissionResponse{
+		Allowed: permitted,
+		Result: &metav1.Status{
+			Message: msg,
+		},
+	}
+}
+
+func validatePodchaos(rawObj []byte) (bool, string, error) {
```

Comment:
Should we moidfy the development guide to show that we can register validate hook here?

Reply:
It's ok to me.

Reply:
Can we implement different chaos validation functions in the corresponding chaos definition?

Reply:
> Can we implement different chaos validation functions in the corresponding chaos definition?

I didn't get your point. Can you provide more detail?

Reply:
eg: in `api/v1alpha1/iochaos_types.go` 
```
func (i *IOChaos) Validate{
xxxxx
}
```

Reply:
we can define an `interface` with `Validate ` function for all Chaos  and require all Chaos struct to implement this interface . Here we just call this method and don't care about the specific type of chaos and the specific implementation

Reply:
@Yisaer PTAL again

Reply:
updated.